azurerm/resource_arm_template_deployment.go

```diff
@@ -231,13 +231,35 @@ func normalizeJson(jsonString interface{}) string {
 	return string(b[:])
 }
 
-func templateDeploymentStateRefreshFunc(client *ArmClient, resourceGroupName string, name string) resource.StateRefreshFunc {
+func waitForTemplateDeploymentToBeDeleted(ctx context.Context, client resources.DeploymentsClient, resourceGroup, name string) error {
+	// we can't use the Waiter here since the API returns a 200 once it's deleted which is considered a polling status code..
```

Comment:
The 200 here is frustrating

Reply:
yeah - I've taken another look into this this morning verifying as I thought this could be the result of a separate bug - but I'm seeing the same error so ¯\_(ツ)_/¯